### PR TITLE
Feature/config override secrets

### PR DESCRIPTION
**Pull request by pingers**
Adds project defaults and environment specific configuration overrides via secrets.

Secrets get mounted in /etc/secret-environment and /etc/secret-project - it's then up to the applications to read and respect precedence of those secrets.

### CODE REVIEW

File: web/modules/custom/shepherd/shp_orchestration/src/Service/Configuration.php [hunk 1]
```diff
@@ -128,7 +128,21 @@ public function extractEnvironmentVariables(NodeInterface $node) {
    *   An array of secrets.
    */
   public function getSecrets(NodeInterface $node) {
-    return $this->moduleHandler->invokeAll('shp_secrets', [$node]);
```
Comment (pingers) [inline web/modules/custom/shepherd/shp_orchestration/src/Service/Configuration.php]: Removed this as there's no implementations and doesn't make a lot of sense now that we can just attach secrets to deployments.

Comment (singularo) [reply web/modules/custom/shepherd/shp_orchestration/src/Service/Configuration.php]: :+1:

File: web/modules/custom/shepherd/shp_orchestration/src/Plugin/OrchestrationProvider/OpenShiftOrchestrationProvider.php [hunk 2]
```diff
@@ -324,7 +325,7 @@ public function updatedEnvironment(
     $deployment_name = self::generateDeploymentName($environment_id);
     $formatted_env_vars = $this->formatEnvVars($environment_variables, $deployment_name);
 
-    if (!$volumes = $this->setupVolumes($project_name, $deployment_name)) {
+    if (!$volumes = $this->setupVolumes($project_name, $deployment_name, $storage_class, $secrets)) {
```
Comment (singularo) [inline web/modules/custom/shepherd/shp_orchestration/src/Plugin/OrchestrationProvider/OpenShiftOrchestrationProvider.php]: That was missing? weird.

Comment (singularo) [reply web/modules/custom/shepherd/shp_orchestration/src/Plugin/OrchestrationProvider/OpenShiftOrchestrationProvider.php]: storage class, I mean

Comment (pingers) [reply web/modules/custom/shepherd/shp_orchestration/src/Plugin/OrchestrationProvider/OpenShiftOrchestrationProvider.php]: Ya, go figure.

**Pull request merged**